content/docs/use/language_support/libc.md

```diff
@@ -0,0 +1,100 @@
+---
+weight: 1
+bookFlatSection: false
+title: "Standard Library (libc) Usage Guidance"
+---
+
+
+# LionsOS libc Integration
+
+This page explains the LionsOS libc implementation and how to configure and
+build it using the provided Makefile snippet.
+
+## Overview
+
+LionsOS uses a custom libc setup based on our
+[fork](https://github.com/au-ts/musllibc/tree/lionsos) of
+[musllibc](https://musl.libc.org/), plus extra components for POSIX support and
+compiler runtime helpers. This setup is compatible with the sDDF build system.
+
+The final libc is built from three main sources:
+
+1. musllibc (our fork): Provides core libc functionality, with system calls
+redirected to a general dispatcher.
+2. Syscall implementations: Functionality found in `lib/libc/posix/`.
+3. Compiler runtime helpers: Low-level arithmetic and runtime support from
+`lib/libc/compiler_rt/`.
+
+## musllibc Fork: Syscall Redirection
+
+Our musllibc fork replaces architecture-specific inline assembly syscall traps
+(like `svc 0` on AArch64) with a generic function call mechanism. Instead of
+invoking syscalls directly with assembly, musl's internal `__syscallN` functions
+now call a dispatcher function via `__sysinfo`.
+
+In upstream musl, `__sysinfo` is normally a hook for vDSO (Virtual Dynamic
+Shared Object) support on Linux systems, letting musl call kernel-provided
+user-space functions without a full syscall. We've repurposed it as a *general*
+syscall dispatcher, since all our syscalls are provided in a user-space library.
+
+As a result, all libc syscall invocations are routed through the LionsOS POSIX
+syscall handler, which is implemented in the `lib/libc/posix/` layer.
+For example:
+
+```c
+#define CALL_SYSINFO(n, ...) ((long(*)(long,...))__sysinfo)(n, ##__VA_ARGS__)
+
+static inline long __syscall3(long n, long a1, long a2, long a3) {
+    return CALL_SYSINFO(n, a1, a2, a3);
+}
+```
+
+LionsOS provides the `__sysinfo` implementation in `libc_init()`:
+
+```c
+void libc_init() {
+    /* Syscall table init */
+    __sysinfo = sel4_vsyscall;
+...
+```
+
+## Usage
+
+To build the LionsOS libc, include the `lib/libc/libc.mk` snippet in your
+top-level Makefile:
+
+```make
```

Comment:
<img width="491" height="63" alt="Image" src="https://github.com/user-attachments/assets/4ddf2b17-c745-4355-a418-f3510b50b8ec" />

Something's weird with the styling for me, is it like this for you to? Maybe worth not having the make specific syntax highlighting...